cmd/pilot-discovery/main.go

```diff
@@ -87,34 +89,30 @@ var (
 
 			stop := make(chan struct{})
 
+			configClient, err := crd.NewClient(flags.kubeconfig, model.ConfigDescriptor{
+				model.RouteRule,
+				model.EgressRule,
+				model.DestinationPolicy,
+			}, flags.controllerOptions.DomainSuffix)
+			if err != nil {
+				return multierror.Prefix(err, "failed to open a config client.")
+			}
+
+			if err = configClient.RegisterResources(); err != nil {
+				return multierror.Prefix(err, "failed to register custom resources.")
+			}
+
 			// Set up values for input to discovery service in different platforms
 			if flags.serviceregistry == platform.KubernetesRegistry || flags.serviceregistry == "" {
-
-				_, client, err := kube.CreateInterface(flags.kubeconfig)
-				if err != nil {
-					return multierror.Prefix(err, "failed to connect to Kubernetes API.")
+				_, client, kuberr := kube.CreateInterface(flags.kubeconfig)
+				if kuberr != nil {
+					return multierror.Prefix(kuberr, "failed to connect to Kubernetes API.")
```

Comment:
Curious why you made this change - 'err' name is used consistently in most go code I've seen.

Reply:
linter is picky about variable shadowing